pandas/io/excel/_openpyxl.py

```diff
@@ -40,6 +40,8 @@ def __init__(
             from openpyxl import load_workbook
 
             self.book = load_workbook(self.handles.handle)
+            self.handles.handle.seek(0)
+            self.handles.handle.truncate()
```

Comment:
that could be a bad idea if later some error happens (user's data is lost). Is there a guarantee that the appended file is always larger then the existing file? Intuitively, that is probaly the case most of the times but there might be cases where that is not correct?

Reply:
might be better to call `self.handles.handle.truncate(self.handles.handle.tell())` in `def close()`.